R/tooltip.R

```diff
@@ -0,0 +1,117 @@
+#' Add a tooltip to a UI element
+#'
+#' Tooltips are useful for showing additional information when focusing (or
+#' hovering over) a UI element.
+#'
+#' @param ... Unnamed arguments can be any valid child of an [htmltools
+#'   tag][htmltools::tags]. Named arguments become HTML attributes on returned
+#'   UI element. Attributes starting with `data-bs-*` may be supplied to further
+#'   customize [tooltip
+#'   options](https://getbootstrap.com/docs/5.3/components/tooltips/).
+#' @param body Content to display in the tooltip.
+#' @param placement The placement of the tooltip relative to the target element.
+#' @param html Whether to treat `body` as HTML. WARNING: setting this `TRUE`
+#'   when the `body` is a function of user `inputs` is dangerous and not
+#'   recommended.
```

Comment:
Could we use `HTML()` or providing a `tag()` or `tagList()` as the signal and avoid this argument?

```r
taglist(
  a(href = "#", "Some text"),
  body = HTML("Tooltip <em>message</em>")
)
```

Is it weird that this would show escaped HTML?

Reply:
Hmm, I'd really like it to be explicit (if someone else's function returns `html`/`shiny.tag`, it wouldn't be very clear what's going on, or how we would opt-out)

Reply:
Do you have a scenario in mind where someone else's function would lead to a surprising result?